core/src/main/java/io/questdb/std/AssociativeCache.java

```diff
@@ -93,15 +108,26 @@ public CharSequence put(CharSequence key, V value) {
 
         if (Chars.equalsNc(key, keys[lo])) {
             if (values[lo] != value) {
-                Misc.free(values[lo]);
+                if (values[lo] == null) {
+                    cachedGauge.inc();
```

Comment:
could we add comments here please, this is hard to follow logic

Reply:
Piece of cake. Added clarifying comments in 83435c8c6393c4086484157fabc24b9920fb36e1